frontend/src/metabase/selectors/routing.ts

```diff
@@ -0,0 +1,6 @@
+import type { State } from "metabase-types/store";
+
+const getRouting = (state: State) => state.routing;
+
+export const getLocation = (state: State) =>
+  getRouting(state).locationBeforeTransitions;
```

Comment:
maybe this one is even cleaner

```suggestion
export const getLocation = (state: State) =>
  state.routing.locationBeforeTransitions;
```